frontend/src/queries/query.ts

```diff
@@ -26,13 +26,16 @@ import {
     isStickinessFilter,
     isTrendsFilter,
 } from 'scenes/insights/sharedUtils'
-import { flattenObject, toParams } from 'lib/utils'
+import { flattenObject, delay, toParams } from 'lib/utils'
 import { queryNodeToFilter } from './nodes/InsightQuery/utils/queryNodeToFilter'
 import { now } from 'lib/dayjs'
 import { currentSessionId } from 'lib/internalMetrics'
 import { featureFlagLogic } from 'lib/logic/featureFlagLogic'
 import { FEATURE_FLAGS } from 'lib/constants'
 
+const QUERY_ASYNC_DELAY_MILLISECONDS = 1000
```

Comment:
This could probably go up to 5 or 15sec without too much worry

Reply:
True, since I now already made sure below that the initial query won't be using that long value 🕺🏻